### PR TITLE
Add 008 network tests 

### PR DESCRIPTION
**Pull request by rvem**
## Description
Problem: New protocol arrived.

Solution: Run network tests on 008 local-chain and edonet.

Based on https://gitlab.com/morley-framework/morley/-/merge_requests/760

## Related issue(s)

https://issues.serokell.io/issue/TM-491

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [x] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

**Comment by rvem** [comment]
:confused: , okay looks like morley-metadata should be updated first

**Comment by rvem** [comment]
Also, looks like we need to export `SenderCap` related stuff from https://gitlab.com/morley-framework/morley/-/blob/master/code/cleveland/src/Morley/Nettest/Caps.hs to reuse them in stablecoin-client

File: haskell/stack.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ extra-deps:
     https://gitlab.com/morley-framework/morley.git
     # ^ CI cannot use ssh, so we use http clone here
   commit:
-    9e69d52741e456bd0976a1ec4c45cc4c43a11722 # master, morley-1.11.1
+    56d74ababeda9aba8e194783fc5e395fa7a6b69a # morley-1.12.0
```
Comment (gromakovsky) [inline haskell/stack.yaml:16]: I don't know if we still support `tezos-client-7.5` after this change, but in https://github.com/tqtezos/stablecoin/blob/master/haskell/README.md we have
> tezos-client executable must be available and is assumed to be in $PATH by default.

and I think it should be updated to require 8.0 or newer

Comment (rvem) [reply haskell/stack.yaml]: Updated README

File: haskell/README.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ Otherwise, you should prefix all commands with `stack run` and put `--` after `s
 
 The help message should be sufficiently descriptive, however, there are some caveats:
 1. [`tezos-client`](http://tezos.gitlab.io/introduction/howtoget.html) executable must be available and is assumed to be in `$PATH` by default.
+   Note that only `tezos-client` from version v8.0 is supported.
```
Comment (pasqu4le) [inline haskell/README.md:22]: Nitpick rephrasing.
```suggestion
   Note that only `tezos-client` versions v8.0 or above are supported.
```

File: haskell/nettest/Nettest.hs [hunk 3]
```diff
@@ -107,16 +107,17 @@ scNettestScenario originateTransferlist transferlistType = uncapsNettest $ do
...
       -> capsM ()
     callTransfers = mapM_ $ \(from@(arg #from_ -> from_), destinations) ->
-      callFrom
-        (AddressResolved from_)
+      withSender
+        (AddressResolved from_) $
+        call
         sc
         (Call @"Transfer")
         (constructTransfersFromSender from destinations)
```
Comment (pasqu4le) [inline haskell/nettest/Nettest.hs:123]:
```suggestion
      withSender (AddressResolved from_) $
        call sc (Call @"Transfer") (constructTransfersFromSender from destinations)
```

File: haskell/nettest/Nettest.hs [hunk 4]
```diff
@@ -126,24 +127,27 @@ scNettestScenario originateTransferlist transferlistType = uncapsNettest $ do
...
         sc'
         (Call @"Configure_minter")
         (ConfigureMinterParam for expectedAllowance newAllowance)
 
     removeMinter :: Address -> Address -> capsM ()
     removeMinter from whom =
-      callFrom
-        (AddressResolved from)
+      withSender
+        (AddressResolved from) $
```
Comment (pasqu4le) [inline haskell/nettest/Nettest.hs:140]: So... there's a bunch more cases of the two suggestions above in this file :sweat_smile: 

The suggestion is to move the first argument of `withSender` to its line and add indentation to the arguments of `call`.
Note that in many cases where now `call`  can fit with its arguments on a single line of ~80 and I think we should do that where possible.

File: haskell/nettest/Nettest.hs [hunk 5]
```diff
@@ -126,24 +127,27 @@ scNettestScenario originateTransferlist transferlistType = uncapsNettest $ do
...
     configureMinter' :: TAddress Parameter -> Address -> Address -> Maybe Natural -> Natural -> capsM ()
     configureMinter' sc' from for expectedAllowance newAllowance =
-      callFrom
-        (AddressResolved from)
+      withSender
+        (AddressResolved from) $
+        call
         sc'
         (Call @"Configure_minter")
         (ConfigureMinterParam for expectedAllowance newAllowance)
```
Comment (pasqu4le) [inline haskell/nettest/Nettest.hs:135]:
```suggestion
      withSender (AddressResolved from) $
        call
          sc'
          (Call @"Configure_minter")
          (ConfigureMinterParam for expectedAllowance newAllowance)
```

**Review by pasqu4le — COMMENTED** [review]
Minor suggestions.

File: haskell/nettest/Nettest.hs (hunk 4 above)
Comment (rvem) [reply haskell/nettest/Nettest.hs]: Looks like I've updated everything now

**Comment by pasqu4le** [comment]
Ok one last thing to do after [the morley MR](https://gitlab.com/morley-framework/morley/-/merge_requests/760) gets merged:

in `Stablecoin.Client.Parser` we re-define `addressOrAliasOption`, which should no longer be necessary after that, so let's removeit and import the function from `morley-client` instead.

Aside from that all LGTM, so feel free to squash already when you update the `stack.yaml` commit again.

**Review by pasqu4le — APPROVED** [review]
All LGTM now :ok_hand:

**Pull request merged**